### PR TITLE
Use `amount.Amount` type in `Archive` interface

### PR DESCRIPTION
**Pull request by Mike-CZ**
This PR uses `amount.Amount` type in `Archive` interface. Should be merged after https://github.com/Fantom-foundation/Carmen/pull/920.

### CODE REVIEW

**Review by HerbertJordan — DISMISSED** [review]
Looks good, thanks!

File: go/backend/archive/ldb/archive.go [hunk 1]
```diff
@@ -239,18 +242,18 @@ func (a *Archive) Exists(block uint64, account common.Address) (exists bool, err
...
 	it := a.db.NewIterator(&keyRange, nil)
 	defer it.Release()
 
 	if it.Next() {
-		copy(balance[:], it.Value())
+		balance = amount.NewFromBytes(it.Value()[:]...)
 		return balance, nil
 	}
-	return common.Balance{}, it.Error()
+	return amount.New(), it.Error()
```
Comment (kjezek) [inline go/backend/archive/ldb/archive.go:256]: should not we have `amount.Zero()` ?

**Review by kjezek — APPROVED** [review]
thanks, it looks good

File: go/backend/archive/ldb/archive.go (hunk 1 above)
Comment (Mike-CZ) [reply go/backend/archive/ldb/archive.go:256]: Once it's all merged, I'll create a new PR with `Zero` factory method and all `New()` instances replaced.

**Review by kjezek — APPROVED** [review]

**Review by kjezek — APPROVED** [review]

**Pull request merged**